aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1799,7 +1799,9 @@ Tensor sparse_compressed_to_sparse(const Tensor& self, int64_t sparse_dim) {
         .flatten(-2, -1);
 
       values = self.values().flatten(0, 2);
-      coalesced = nnz == 1;
+
+      // Not spanning across several rows produces coalesced results.
+      coalesced |= (blocksize[0] == 1);
```

Comment:
Notice that the condition `blocksize[0] == 1` does not apply to the BSC case, so we'll need
```suggestion
      coalesced |= (the_layout == kSparseBsr && blocksize[0] == 1);
```
or similar.

Since this issue has not been caught up by tests, I think we'll still need to have more rigorous testing here.

Reply:
Having upper-lower block-diagonal matrices is a very nice candidate.